### ISSUE TITLE
Pylance false positive for missing method on a wrapt.ObjectProxy subclass

### ISSUE THREAD

**Issue by ddelange** [microsoft/pyright#11295]
Pylance doesn't seem to understand [`wrapt.ObjectProxy`](https://wrapt.readthedocs.io/en/master/wrappers.html) subclasses. [wrapt](https://pypi.org/project/wrapt/) is one of the most popular python libs ([290M](https://pypistats.org/packages/wrapt) monthly downloads).

## Environment data

-   Pylance version: latest
-   OS and version: irrelevant
-   Python version (& distribution if applicable, e.g. Anaconda): irrelevant

## Code Snippet

```python
from smart_open import open

with open("example.txt") as f:
    for line in f:
        print(line.strip())
```

With `smart_open` 7.5.0:
<img width="840" height="185" alt="error on open with smart_open 7.5.0" src="https://github.com/user-attachments/assets/f95e20d8-775c-4022-be34-2812b48bda61" />

With `smart_open` 7.5.1:
<img width="262" height="112" alt="no error on open with smart_open 7.5.1" src="https://github.com/user-attachments/assets/0d7e21ca-b472-4611-b30e-cdb9fe4b9e50" />

Potentially relevant VS Code workspace settings:

```json
{
  "python.languageServer": "Pylance",
  "python.analysis.typeCheckingMode": "standard",
  "python.analysis.indexing": true
}
```

## Repro Steps

1. run `pip install smart_open==7.5.0`
2. observe screenshot 1
3. run `pip install smart_open==7.5.1`
4. observe screenshot 2

The difference between the two versions is a band-aid PR specifically for pylance (the diff is just to make pylance happy, otherwise unnecessary): https://github.com/piskvorky/smart_open/pull/917

This is the subclass in question: https://github.com/piskvorky/smart_open/blob/v7.5.0/smart_open/utils.py#L210

 _Originally posted by @bvobart in https://github.com/piskvorky/smart_open/issues/916#issuecomment-3946302359_

**Comment by brianschubert** [comment]
`wrapt` doesn't define `__enter__` in it's stubs, so pyright is correct to flag this:
https://github.com/GrahamDumpleton/wrapt/blob/27c5248f43f1b248274922ab4a0224fa90a45bab/src/wrapt/__init__.pyi#L42-L47

You'll need to report this to wrapt.

**Comment by ddelange** [comment]
Hi @brianschubert 👋 

Thanks for the quick reply. I've opened https://github.com/GrahamDumpleton/wrapt/issues/314. Could you add additional TODOs there if applicable? I'll close this ticket in favor of the wrapt ticket.
